api/v1alpha1/etcdpeer_types.go

```diff
@@ -66,6 +66,11 @@ type EtcdPeerSpec struct {
 	// Storage is the configuration of the disks and mount points of the Etcd
 	// pod.
 	Storage *EtcdPeerStorage `json:"storage,omitempty"`
+
+	// Decommissioned is set to true if this EtcdPeer has been removed from an etcd cluster
+	// and its data can be safely deleted.
+	// +optional
+	Decommissioned bool `json:"decommissioned"`
```

Comment:
What's the purpose/requirement for this field? I would expect the PVC to be deleted after:

1) the EtcdCluster controller has called MemberRemove
2) the EtcdCluster controller has deleted the EtcdPeer resource
3) The PVC can be deleted.

I'm really uncertain about this field and it really feels like it should not exist personally. As an alternative, could the EtcdCluster controller just add the *finalizer* to the EtcdPeer before deleting it, and then have the EtcdPeer controller delete the PVC for any EtcdPeer resource that is (1) deleted and (2) has the finalizer?

This would *also* allow us to control when the finalizer is set from the EtcdCluster resource, i.e. the EtcdCluster controller could choose *not* to set the finalizer if the user has requested to never delete PVCs.